compiler/qsc_partial_eval/src/lib.rs

```diff
@@ -1905,6 +1975,129 @@ impl<'a> PartialEvaluator<'a> {
         }
     }
 
+    #[allow(clippy::too_many_lines)]
+    fn generate_instructions_for_binary_operation_with_double_operands(
+        &mut self,
+        bin_op: BinOp,
+        lhs_operand: Operand,
+        rhs_operand: Operand,
+        bin_op_expr_span: PackageSpan, // For diagnostic purposes only.
+    ) -> Result<rir::Variable, Error> {
+        let rir_variable = match bin_op {
+            BinOp::Add => {
+                let bin_op_variable_id = self.resource_manager.next_var();
+                let bin_op_rir_variable = rir::Variable::new_double(bin_op_variable_id);
+                let bin_op_rir_ins =
+                    Instruction::Fadd(lhs_operand, rhs_operand, bin_op_rir_variable);
+                self.get_current_rir_block_mut().0.push(bin_op_rir_ins);
+                bin_op_rir_variable
```

Comment:
Agree with clippy that this function is too long. A low hanging fruit is that 3/5 lines in all of these arms (except division) are identical and can live outside the `match`. (For division, I don't think it matters if you do the divide-by-zero check after you've already allocated the next variable, so I think that could stay inside the `match`?)